### DISCUSSION TITLE
RS485 Renogy battery

### DISCUSSION THREAD

**Discussion by janderholm** [Louisvdw/dbus-serialbattery#196] [Q&A] [answered]
I'm having difficulty tracking down how the RS485 port of Renogy batteries are wired. I have the Victron USB - RS485 cable but can't find how to connect it to the battery.

Does anyone have a definitive answer?

**Comment by janderholm** [comment]
I opened the battery and found a TI chip marked VP485, which I believe is this RS485 transceiver:
https://www.ti.com/lit/ds/symlink/sn65hvd485e.pdf?ts=1662566728632

I could trace the "A" pin to the 7th RJ45 pin, and the "B" pin to the 6th. The 5th seem to be connected to ground.
The numbers are the usual RJ45 numbers:
https://www.showmecables.com/blog/post/rj45-pinout

The Victron RS485 cable seem to be one of these:
https://www.farnell.com/datasheets/652302.pdf

I.e:
Pin 5 - GND - Black
Pin 6 - B - Yellow
Pin 7 - A - Orange

Unfortunately it does't work, and neither does switching A and B, so I'm not sure what's going on.

**Comment by janderholm** [comment, marked as answer]
I found an excellent picture, although the pin numbering is reversed compared to how I did it above. And there does not seem to be any 5V pin on the Renogy battery.
https://github.com/rosswarren/renogymodbus

![Image showing the pins of an RJ45 connector for Renogy devices](https://user-images.githubusercontent.com/613642/179362448-12a805d1-4475-45cc-b3d7-c8a8e9c4b409.png "Renogy RJ45 RS485 interface")

The pins match how I connected my cable, but unfortunately still no communications with the battery.

**Reply by mark-1003** [reply]
I use a USB-to-RS485 converter with a CH340 chip and connected the wiring according the image above.
This works with my RBT100LFP12SH battery.

**Comment by janderholm** [comment]
Finally I've managed to talk to my battery using the tool in the link above. The following command shows that communication is possible:

$ renogymodbus --device smart_battery --portname /dev/ttyUSB0 --slaveaddress 247

I created the following issue to track this. I'll try to get it to work. 

https://github.com/Louisvdw/dbus-serialbattery/issues/197

**Comment by Louisvdw** [comment]
The important thing to remember if you are using a Victron USB - RS485 cable is that this is a isolate cable and requires that you supply both sides with power. The USB side will get power from USB, but you do have to use the red 5V+ wire as well.

You also need to look at the log files and see if there is anything that will help. See the Troubleshoot section in the Wiki

**Reply by janderholm** [reply]
I'm using this cable:
https://www.victronenergy.com/accessories/rs485-to-usb-interface
https://www.victronenergy.com/live/energy-meters:start#faq

And it's not isolated so there shouldn't be a problem:

> Q10: Can I use an isolated USB-RS485 interface?
>
> Yes. The interfaces we sell are non isolated; suitable for most use cases. 

I don't think Victron sell isolated RS485 interfaces. The VE.Direct - USB cable is isolated, and I believe the MK3 - USB might be as well. This cable isn't of their production but a more generic brand.

Anyway, from my measurements this battery doesn't even expose a 5V power supply pin. But I never checked if pin 8 was connected to the PCB so I can't be sure.

However, I didn't dig into it further so take this with a huge grain of salt, but it does not seem like the ground on the RJ45 port is connected directly to the negative of the battery. Don't know what's in between.
